### PR TITLE
feat: add a gradle plugin

### PR DESCRIPTION
**Pull request by tagoro9**
**Description**

Refactor a lot of the code into some utils (don't like the _config file name) and add a gradle preset.

**Changes**

* build(deps): add gradle-semantic-release-plugin
* refactor: extract at end plugins into separate file
* refactor: extract semantic-release/git into function
* refactor: add a createPreset function
* test(npm): fix typo
* test: move some tests from npm to config
* feat: create a gradle preset

🚀 PR created with [fotingo](https://github.com/tagoro9/fotingo)

### CODE REVIEW

File: src/_config.ts [hunk 1]
```diff
@@ -20,3 +26,35 @@ export function semanticReleaseGit(assets: string[]) {
     },
   ];
 }
+
+/**
+ * Given a semantic release plugin config, return the plugin name
+ * @param p Plugin config
```
Comment (dross15) [inline src/_config.ts:32]: do we want to use a more verbose name than a singular letter?

**Review by dross15 — APPROVED** [review]

File: test/gradle.test.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import npm from "~/npm";
```
Comment (Toolo) [inline test/gradle.test.ts:1]: Shouldn't this import from `gradle`?

**Review by Toolo — APPROVED** [review]
Just a comment to review the test import 😄

**Review by tagoro9 — COMMENTED** [review]
I had forgotten to post my self review

**Pull request merged**